shell/platform/android/flutter_main.cc

```diff
@@ -104,33 +104,27 @@ void ConfigureShorebird(std::string android_cache_path,
   fml::CreateDirectory(fml::paths::GetCachesDirectory(), {"shorebird_updater"},
                        fml::FilePermission::kReadWrite);
 
-  AppParameters app_parameters;
-  app_parameters.release_version = version.c_str();
-  app_parameters.cache_dir = cache_dir.c_str();
-
-  // Converting the strings in `settings.application_library_path` to `char*`.
-  // We maintain a separate vector so that the lifetime of the `char*` is clear.
-  // If we used `c_str` then we would have to be very careful not to modify the
-  // string or have taken a copy of the string as `c_str` is only valid for the
-  // lifetime of the string.
-  std::vector<std::vector<char>> libapp_paths_storage;
-  std::vector<const char*> original_libapp_paths;
-  for (auto& path : settings.application_library_path) {
-    libapp_paths_storage.push_back(std::vector<char>(path.begin(), path.end()));
-    original_libapp_paths.push_back(libapp_paths_storage.back().data());
-  }
+  // Using a block to make AppParameters lifetime explicit.
+  {
+    AppParameters app_parameters;
+    app_parameters.release_version = version.c_str();
+    app_parameters.cache_dir = cache_dir.c_str();
+
+    // https://stackoverflow.com/questions/26032039/convert-vectorstring-into-char-c
+    std::vector<const char*> c_paths{};
+    for (const auto& string : settings.application_library_path) {
+      c_paths.push_back(string.c_str());
```

Comment:
wouldn't this have the same problem we originally had since this is pushing a reference to the path in `settings.application_library_path` instead of creating a copy of the string?

Reply:
Yes, but "string" here is intentionally a reference into the vector of strings.  I think last time we ended up with a temp/copy of the path from which we took a c_str() which then became invalid when the temp went out of scope?

Reply:
Oh yeah I see...nvm makes sense